lib/grammers-mtproto/src/mtp/encrypted.rs

```diff
@@ -1159,19 +1159,27 @@ impl Mtp for Encrypted {
         // Check to see if the next salt can be used already. If it can, drop the current one and,
         // if the next salt is the last one, fetch more.
         if let Some((start_secs, start_instant)) = self.start_salt_time {
-            if let Some(salt) = self.salts.get(self.salts.len() - 2) {
-                let now = start_secs + start_instant.elapsed().as_secs() as i32;
-                if now >= salt.valid_since + SALT_USE_DELAY {
-                    self.salts.pop();
-                    if self.salts.len() == 1 {
-                        info!("only one future salt remaining; asking for more salts");
-                        let body = tl::functions::GetFutureSalts {
-                            num: NUM_FUTURE_SALTS,
+            let should_fetch_salts = self.salts.len() < 2
+                || 'value: {
```

Comment:
This was a hackjob on my part. I guess we can keep it, but it's definitely not the cleanest.